flowmachine/flowmachine/features/subscriber/label_event_score.py

```diff
@@ -0,0 +1,192 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# # License, v. 2.0. If a copy of the MPL was not distributed with this
+# # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from shapely.geometry import shape
+from shapely.geometry.base import BaseGeometry
+from typing import List, Dict, Union, Any
+
+from .scores import EventScore
+from .subscriber_location_cluster import _JoinedHartiganCluster
+from ...core import Query
+
+
+class LabelEventScore(Query):
+    """
+    Represents a label event score class.
+
+    This class will label a table containing scores based on a labelling
+    dictionary. It allows one to specify labels which every subscriber must have.
+    This class is used to label locations based on scoring signatures in the
+    absence of other automated labelling mechanisms.
+
+    Parameters
+    ----------
+    scores : flowmachine.Query
+        A flowmachine.Query object.
+        This represents a table that contains scores which are used to label
+        a given location. This table must have a subscriber column (called subscriber).
+    labels : dict
+        A dictionary whose keys are the label names and the values geojson shapes,
+        specified hour of day, and day of week score, with hour of day score on the x axis
+        and day of week score on the y axis, where all values are in the range [-1, 1]
+    required :
+        Specifies a label which every subscriber must possess independently of
+        the score.  This is used in cases where, for instance, we require
+        that all subscribers must have an evening/home location.
+    """
+
+    def __init__(
+        self,
+        *,
+        scores: Union[EventScore, _JoinedHartiganCluster],
+        labels: Dict[str, Dict[str, Any]] = {
+            "evening": {
+                "type": "MultiPolygon",
+                "coordinates": [
+                    [[[0.000_001, 0.5], [0.000_001, 1], [1, 1], [1, 0.5]]],
+                    [[[0.000_001, -1], [0.000_001, -0.5], [1, -0.5], [1, -1]]],
+                ],
+            },
+            "day": {
+                "type": "Polygon",
+                "coordinates": [[[-1, -0.5], [-1, 0.5], [0, 0.5], [0, -0.5]]],
+            },
+        },
+        required: Union[str, None] = None,
+    ):
+
+        self.scores = scores
+        if not isinstance(scores, Query):
+            raise TypeError(
+                "Scores must be of type Query, e.g. EventScores, Table, CustomQuery"
+            )
+
+        labels = LabelEventScore._make_bounds_dict(labels)
+        LabelEventScore.bounds_dict_has_overlaps(labels)
```

Comment:
This reads a little confusingly, because the method name implies that it returns a boolean but the return value is ignored here. Can we rename it to `LabelEventScore.verify_bounds_dict_has_no_overlaps(labels)` or similar?